MDANSE/Src/MDANSE/Trajectory/MdanseTrajectory.py

```diff
@@ -507,109 +524,128 @@ def has_variable(self, variable: str) -> bool:
+                == atom_property.encode("utf-8"),
             )[0]
             if len(temp) == 0:
-                if property_name == "dummy":
+                if atom_property == "dummy":
                     try:
-                        return ATOMS_DATABASE.get_atom_property(symbol, property_name)
+                        return ATOMS_DATABASE.get_atom_property(symbol, atom_property)
                     except KeyError:
                         if (
                             "_" in symbol
                         ):  # this is most likely an artificial atom from a molecule
                             return 0  # the molecule atoms are not dummy
                 else:
                     raise KeyError(
-                        f"Property {property_name} is not in the trajectory's internal database."
+                        f"Property {atom_property} is not in the trajectory's"
+                        " internal database.",
                     )
             index = temp.flatten()[0]
-            self._property_map[property_name] = index
-        index = self._property_map[property_name]
+            self._property_map[atom_property] = index
+        index = self._property_map[atom_property]
         if index not in self._data_types:
             self._data_types[index] = self._h5_file["/atom_database/property_types"][
                 index
             ]
         data_type = self._data_types[index]
+        if index not in self._data_units:
+            data_unit = "none"
+            try:
+                unit_lookup = self._h5_file["/atom_database/property_units"]
+            except KeyError:
+                if not self.warned_about_complex_numbers:
+                    LOG.warning(
+                        "This trajectory file was generated with old MDANSE. If you"
+                        "need complex b, please generate it again.",
+                    )
+                    self.warned_about_complex_numbers = True
+            else:
+                data_unit = unit_lookup[index]
+            self._data_units[index] = data_unit
+        data_unit = self._data_units[index]
         if (symbol, index) not in self._property_cache:
             value = self._h5_file[f"/atom_database/{symbol}"][index]
             if data_type != b"complex":
                 value = value.real
             self._property_cache[(symbol, index)] = value
         value = self._property_cache[(symbol, index)]
-        if property_name == "color":
-            value = float(value)
-            num1 = round(value // 0x10000)
-            num2 = round((value - num1 * 0x10000) // 0x100)
-            num3 = round(value - num1 * 0x10000 - num2 * 0x100)
-            return ";".join([str(int(x)) for x in [num1, num2, num3]])
+        if atom_property == "color":
+            value = str_to_num(value)
+            r, value = divmod(value, 65536)
+            g, b = divmod(value, 256)
+            return ";".join([str(int(x)) for x in [r, g, b]])
```

Comment:
Realised we can simplify this further:
```suggestion
            return ";".join(map(str, value.to_bytes(3, 'big')))
```